futurex_openedx_extensions/helpers/clickhouse_operations.py

```diff
@@ -0,0 +1,218 @@
+"""Clickhouse helper functions."""
+from __future__ import annotations
+
+import os
+from typing import Any
+
+import yaml  # type: ignore
+from clickhouse_connect import get_client as clickhouse_get_client
+from clickhouse_connect.driver.httpclient import HttpClient
+from clickhouse_connect.driver.query import QueryResult
+from django.conf import settings
+from django.core.paginator import EmptyPage
+
+
+class ClickhouseBaseError(Exception):
+    """Clickhouse base error."""
+
+
+class ClickhouseClientNotConfiguredError(ClickhouseBaseError):
+    """Clickhouse client not configured error."""
+
+
+class ClickhouseClientConnectionError(ClickhouseBaseError):
+    """Clickhouse client connection error."""
+
+
+class ClickhouseQueryParamsError(ClickhouseBaseError):
+    """Clickhouse query parameters error."""
+
+
+class ClickhouseDefaultQueriesError(ClickhouseBaseError):
+    """Clickhouse default queries error."""
+
+
+def get_client() -> HttpClient:
+    """
+    Get Clickhouse client.
+
+    :return: Clickhouse client.
+    :rtype: HttpClient
+    """
+    try:
+        username = settings.FX_CLICKHOUSE_USER
+        password = settings.FX_CLICKHOUSE_PASSWORD
+    except AttributeError as exc:
+        raise ClickhouseClientNotConfiguredError(f'Error getting Clickhouse credentials: {exc}') from exc
+
+    try:
+        client = clickhouse_get_client(host='clickhouse', port=8123, username=username, password=password)
+    except Exception as exc:
+        raise ClickhouseClientConnectionError(f'Error getting Clickhouse client: {exc}') from exc
+
+    return client
+
+
+def escape_param(param_name: str, param_value: Any) -> str:
```

Comment:
Isn't that part of the client python funcitons?

Why are we re-implemting these essential functions?


```
client.query('SELECT * FROM {table:Identifier} WHERE date >= {v1:DateTime} AND string ILIKE {v2:String}', parameters=parameters)
```

https://github.com/ClickHouse/clickhouse-connect/blob/1550e8976fcad232f6d2ca873e91f86db8e3b7d5/clickhouse_connect/driver/query.py#L407-L412

Kindly be mindful of the maintainince cost of such functions.

Reply:
thank you! `@thraxil` would be happy if he was working on this project :)

Reply:
stale, my bad. I've removed it. Thank you for the check